### DISCUSSION TITLE
Possible to have window messages alongside dinput8 usage?

### DISCUSSION THREAD

**Discussion by ThirteenAG** [samuelgr/Xidi#38] [Q&A]
I noticed that when dinput8 creates a device, such as mouse, the window stops receiving WM_INPUT messages for that device. Is it possible to have both?

**Comment by samuelgr** [comment]
Is this behavior specific to Xidi or is it true for DirectInput generally? From [this link](https://learn.microsoft.com/en-us/windows/win32/dxtecharts/taking-advantage-of-high-dpi-mouse-movement#directinput) it looks like internally DirectInput consumes WM_INPUT messages, so I suspect probably the latter.

If I understand correctly, it looks like WM_INPUT is for raw input, and applications won't receive WM_INPUT unless they explicitly [register for it from specific devices](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerrawinputdevices#remarks). So it is possible that an application registers for raw input, then creates a DirectInput device object for the same device as is registered for raw input, at which point DirectInput gets in the way (though I am not by any means an expert on handling raw input).

**Comment by ThirteenAG** [comment]
> DirectInput device object for the same device as is registered for raw input, at which point DirectInput gets in the way (though I am not by any means an expert on handling raw input).

Yeah, that seems to be the case, because I only was able to get raw input by calling RegisterRawInputDevices in a loop, or by disabling dinput device creation. Hence wondering if possible to do a passthrough of some sort, or not.
